frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx

```diff
@@ -21,9 +22,14 @@ export const CreateApiKeyModal = ({ onClose }: { onClose: () => void }) => {
   const secretKey = response?.data?.unmasked_key || "";
 
   const handleSubmit = useCallback(
-    async vals => {
-      await createApiKey(vals);
-      setModal("secretKey");
+    async (vals: { group_id: number | null; name: string }) => {
+      // bad
+      if (vals.group_id !== null) {
+        await createApiKey({
+          ...vals,
+        } as CreateApiKeyRequest);
```

Comment:
why spreading here? if it's for type reasons you can sometimes do `vals as unknown as CreateApiKeyRequest`

Reply:
You're right. This was from me trying  a couple things, but spreading isn't needed. Will fix